core/drivers/stm32_rtc.c

```diff
@@ -778,7 +1073,28 @@ static TEE_Result stm32_rtc_probe(const void *fdt, int node,
 	res = stm32_rtc_init();
 	if (res)
 		return res;
-	rtc_register(&stm32_rtc);
+
+	if (IS_ENABLED(CFG_DRIVERS_RTC))
+		rtc_register(&stm32_rtc);
```

Comment:
```suggestion
	rtc_register(&stm32_rtc);
```
is enough.

Reply:
In  **[core/include/drivers/rtc.h](https://github.com/OP-TEE/optee_os/blob/4.6.0/core/include/drivers/rtc.h#L191-L193)**
```c
#ifdef CFG_DRIVERS_RTC
(snip)
/* Register a RTC device as the system RTC */
void rtc_register(struct rtc *rtc);
(snip)
#else

static inline void rtc_register(struct rtc *rtc __unused) {}
(snip)
#endif
```